### PR TITLE
Create component metadata for pubsub.kafka

### PR DESCRIPTION
**Pull request by SpikeWong**
#2579

### CODE REVIEW

File: pubsub/kafka/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+version: v1
+status: stable
+title: "Apache Kafka"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-apache-kafka/
+capabilities:
+    - publish
+    - subscribe
+    - query
```
Comment (SpikeWong) [inline pubsub/kafka/metadata.yaml:14]: I don't know the capanolities are correct or not, I could not find a reference to fill with this

File: pubsub/kafka/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+# yaml-language-server: $schema=../../component-metadata-schema.json
+schemaVersion: v1
+type: pubsub
+name: kafka
+version: v1
+status: stable
+title: "Apache Kafka"
```
Comment (SpikeWong) [inline pubsub/kafka/metadata.yaml:7]: Should this be Kafka?

Comment (ItalyPaleAle) [reply pubsub/kafka/metadata.yaml:7]: I think Apache Kafka is fine: https://kafka.apache.org/

File: pubsub/kafka/metadata.yaml (hunk 1 above)
Comment (ItalyPaleAle) [reply pubsub/kafka/metadata.yaml:14]: PubSUb components don't have capabilities, so please just remove the entire section

File: pubsub/kafka/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      example: "my-dapr-app"
+      type: string
+    - name: authRequired
+      required: false
+      description: "Deprecated Enable SASL authentication with the Kafka brokers"
+      example: "false"
+      type: bool
+    - name: authType
+      required: true
+      description: "Configure or disable authentication. Supported values: none, password, mtls, or oidc"
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:38]:
```suggestion
      description: |
        Configure or disable authentication. Supported values: "none", "password", "mtls", or "oidc"
```

File: pubsub/kafka/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      example: "none"
+      type: string
+    - name: saslUsername
+      required: false
+      description: "The SASL username used for authentication. Only required if authType is set to \"password\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:43]:
```suggestion
      description: |
        The SASL username used for authentication. Only required if authType is set to "password"
```

File: pubsub/kafka/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      example: "myuser"
+      type: string
+    - name: saslPassword
+      required: false
+      description: "The SASL password used for authentication. Can be secretKeyRef to use a secret reference. Only required if authType is set to \"password\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:48]:
```suggestion
      description: |
        The SASL password used for authentication. Can be secretKeyRef to use a secret reference. Only required if authType is set to "password"
```

File: pubsub/kafka/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+      example: "mypassword"
+      type: string
+    - name: saslMechanism
+      required: false
+      description: "The SASL Authentication Mechanism you wish to use. Only required if authType is set to \"password\". Defaults to PLAINTEXT"
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:53]:
```suggestion
      description: |
        The SASL Authentication Mechanism you wish to use. Only required if authType is set to "password". Defaults to "PLAINTEXT"
```

File: pubsub/kafka/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      example: "SHA-512"
+      type: string
+    - name: initialOffset
+      required: false
+      description: "The initial offset to use if no offset was previously committed. Should be \"newest\" or \"oldest\". Defaults to \"newest\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:58]:
```suggestion
      description: |
        The initial offset to use if no offset was previously committed. Should be "newest" or "oldest". Defaults to "newest"
```

File: pubsub/kafka/metadata.yaml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+      example: "oldest"
+      type: string
+    - name: maxMessageBytes
+      required: false
+      description: "The maximum size in bytes allowed for a single Kafka message. Defaults to 1024"
+      example: "2048"
+      type: number
+    - name: consumeRetryInterval
+      required: false
+      description: "The interval between retries when attempting to consume topics. Treats numbers without suffix as milliseconds. Defaults to 100ms"
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:68]:
```suggestion
      description: |
        The interval between retries when attempting to consume topics. Treats numbers without suffix as milliseconds. Defaults to "100ms"
```

File: pubsub/kafka/metadata.yaml [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+      example: "200ms"
+      type: string
+    - name: consumeRetryEnabled
+      required: false
+      description: "Disable consume retry by setting \"false\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:73]:
```suggestion
      description: |
        Disables consumer retry by setting this to "false"
```

File: pubsub/kafka/metadata.yaml [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+      example: "true"
+      type: bool
+    - name: version
+      required: false
+      description: "Kafka cluster version. Defaults to 2.0.0.0"
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:78]:
```suggestion
      description: |
        Kafka cluster version. Defaults to "2.0.0.0"
```

File: pubsub/kafka/metadata.yaml [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+      example: "-----BEGIN CERTIFICATE-----\n<base64-encoded DER>\n-----END CERTIFICATE-----"
+      type: string
+    - name: clientKey
+      required: false
+      description: "Client key, required for authType mtls. Can be secretKeyRef to use a secret reference"
+      example: "-----BEGIN RSA PRIVATE KEY-----\n<base64-encoded DER>\n-----END RSA PRIVATE KEY-----"
+      type: string
+    - name: skipVerify
+      required: false
+      description: "Skip TLS verification, this is not recommended for use in production. Defaults to \"false\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:98]:
```suggestion
      description: |
        Skip TLS verification, this is not recommended for use in production. Defaults to "false"
```

File: pubsub/kafka/metadata.yaml [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+      example: "true"
+      type: bool
+    - name: disableTls
+      required: false
+      description: "Disable TLS for transport security. To disable, you’re not required to set value to \"true\". This is not recommended for use in production. Defaults to "false""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:103]:
```suggestion
      description: |
        Disable TLS for transport security. This is not recommended for use in production. Defaults to "false", which implies TLS is enabled.
```

File: pubsub/kafka/metadata.yaml [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+      example: "dapr-kafka"
+      type: string
+    - name: oidcClientSecret
+      required: false
+      description: "The OAuth2 client secret that has been provisioned in the identity provider. Required when authType is set to oidc"
+      example: "KeFg23!"
+      type: string
+    - name: oidcScopes
+      required: false
+      description: "Comma-delimited list of OAuth2/OIDC scopes to request with the access token. Recommended when authType is set to oidc. Defaults to \"openid\""
```
Comment (ItalyPaleAle) [inline pubsub/kafka/metadata.yaml:123]:
```suggestion
      description: |
        Comma-delimited list of OAuth2/OIDC scopes to request with the access token. Recommended when authType is set to oidc. Defaults to "openid"
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution!

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**